### PR TITLE
Improvement `validate_unique` function when only `PrimaryKey` exists

### PR DESCRIPTION
**Pull request by ever0de**
The existing `validate_unique` function creates a unique constraint by looping through all rows even when only `PrimaryKey` exists.

Use `Store::fetch_data` to improve speed in the above situation.

### CODE REVIEW

File: core/src/executor/validate.rs [hunk 1]
```diff
@@ -74,39 +79,93 @@ impl UniqueConstraint {
...
+                .count();
+
+            match (primary_key_index, other_unique_column_def_count) {
+                (Some(primary_key_index), 0) => Columns::PrimaryKeyOnly(primary_key_index),
+                _ => Columns::All(fetch_all_unique_columns(column_defs)),
+            }
         }
+        ColumnValidation::SpecifiedColumns(column_defs, specified_columns) => Columns::All(
+            fetch_specified_unique_columns(column_defs, specified_columns),
+        ),
```
Comment (ever0de) [inline core/src/executor/validate.rs:127]: `SpecifiedColumns` is for `UPDATE` Statement, but PrimaryKey cannot be updated, so it is not checked.

File: core/src/executor/validate.rs [hunk 2]
```diff
@@ -74,39 +79,93 @@ impl UniqueConstraint {
...
+            for primary_key in row_iter.filter_map(|row| match row.0.get(primary_key_index) {
+                Some(value) => Key::try_from(value).map(Some).transpose(),
+                None => None,
+            }) {
+                let key = primary_key?;
+
+                if storage.fetch_data(table_name, &key).await?.is_some() {
+                    return Err(ValidateError::DuplicateEntryOnPrimaryKeyField(key).into());
+                }
+            }
```
Comment (ever0de) [inline core/src/executor/validate.rs:141]: [futures::join](https://docs.rs/futures/latest/futures/macro.join.html) can be used to improve concurrency.

File: core/src/executor/validate.rs [hunk 3]
```diff
@@ -74,39 +79,93 @@ impl UniqueConstraint {
...
-            .try_for_each(|constraint| {
-                let col_idx = constraint.column_index;
-                let val = row
-                    .get_value(col_idx)
-                    .ok_or(ValidateError::ConflictOnStorageColumnIndex(col_idx))?;
+    match columns {
+        Columns::PrimaryKeyOnly(primary_key_index) => {
+            for primary_key in row_iter.filter_map(|row| match row.0.get(primary_key_index) {
+                Some(value) => Key::try_from(value).map(Some).transpose(),
+                None => None,
```
Comment (ever0de) [inline core/src/executor/validate.rs:134]: If the entered data is not wrong, it must be inaccessible code?

Comment (panarch) [reply core/src/executor/validate.rs:134]: Same code but simple simplification
```rust
for primary_key in row_iter.filter_map(|row| row.0.get(index).map(Key::try_from)) {
    let key = primary_key?;

    if storage.fetch_data(table_name, &key).await?.is_some() {
        return Err(ValidateError::DuplicateEntryOnPrimaryKeyField(key).into());
    }
}
```

**Review by panarch — APPROVED** [review]
Nice performance improvement! Thanks really a lot 👍 👍

**Pull request merged**